sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -487,6 +487,60 @@ case class GpuInitCap(child: Expression) extends GpuUnaryExpression with Implici
     }
 }
 
+case class GpuStringRepeat(input: Expression, repeatTimes: Expression)
+    extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+  override def left: Expression = input
+  override def right: Expression = repeatTimes
+  override def dataType: DataType = input.dataType
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, IntegerType)
+
+  def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector =
+    throw new UnsupportedOperationException(s"Cannot columnar evaluate expression: $this")
+
+  def doColumnar(input: GpuColumnVector, repeatTimes: GpuColumnVector): ColumnVector = {
+    val repeatTimesCV = repeatTimes.getBase
+
+    // Compute the output size to check for overflow.
+    val outputSizes = input.getBase.repeatStringsSizes(repeatTimesCV)
+    if (outputSizes.getTotalSize > Int.MaxValue) {
+      throw new RuntimeException("Output strings have total size exceed maximum allowed size")
+    }
+
+    // Finally repeat the strings using the pre-computed strings' sizes.
+    input.getBase.repeatStrings(repeatTimesCV, outputSizes.getStringSizes)
+  }
+
+  def doColumnar(input: GpuColumnVector, repeatTimes: GpuScalar): ColumnVector = {
+    repeatTimes.getValue match {
```

Comment:
As mentioned below, this needs to check the validity of the scalar before attempting to access its value and should switch on `repeatTimes.dataType`.

Reply:
I don't think we have to switch on the data type because we are restricting it to ints, but an assertion that it is the expected type would be good.  We do have to check the validity before we pull the data out.